### DISCUSSION TITLE
Is possible to use both sort and drag on table component?

### DISCUSSION THREAD

**Discussion by yuta519** [rsuite/rsuite#2766] [General]
I'm using a table component. 
And I'm trying to use both sort and react-dnd, referring to the docs ([sort](https://rsuitejs.com/components/table/#sort), [dnd](https://rsuitejs.com/components/table/#draggable-with-react-dnd)).

So I added sortable at Column tag on this [Codesandbox](https://codesandbox.io/s/rsuite5-table-with-react-dnd15-ttrsck).

But any sort of icons does not happen. From my research, I think the icons go to the following line like the below screenshot.

![image](https://user-images.githubusercontent.com/12853233/192409391-42b25f59-ac23-4e7a-b1ec-da737c81d0ac.png)

I wrote the issue about this topic.
https://github.com/rsuite/rsuite/issues/2752

**Comment by simonguo** [comment]
DraggableHeaderCell is a custom Cell. The div as a child will fill the cell, causing the sort icon to overflow. You can see the sort icon after you change the div to a span.

```diff
function DraggableHeaderCell({ children, onDrag, id, ...rest }) {
  ...
  return (
    <HeaderCell {...rest} style={{ padding: 0 }}>
-      <div ref={ref} style={styles}>
+      <span ref={ref} style={styles}>
        {children}
-      </div>
+      </span>
    </HeaderCell>
  );
}
```

Or you can adjust the layout of the cell via style.

```diff
function DraggableHeaderCell({ children, onDrag, id, ...rest }) {
   ...
  const styles = {
    padding: "0.6rem 1rem",
    cursor: "grab",
    opacity: isDragging ? 0 : 1,
    borderLeft: isActive ? "2px solid #2589f5" : null,
+   flexGrow: 1
  };

  return (
    <HeaderCell
      {...rest}
      style={{
        padding: 0,
+       display: "flex",
+       alignItems: "center",
+       justifyContent: "space-between"
      }}
    >
      <div ref={ref} style={styles}>
        {children}
      </div>
    </HeaderCell>
  );
}
```
Complete example: https://codesandbox.io/s/rsuite5-table-with-react-dnd15-ttrsck
